linesearch.go

```diff
@@ -62,6 +62,11 @@ func (ls *Linesearch) Iterate(loc *Location, xNext []float64) (EvaluationType, I
 		return NoEvaluation, ls.iterType, nil
 	}
 	if ls.iterType == MajorIteration {
+		// Compare the starting points for the previous and next iteration to
```

Comment:
This doesn't seem like the right to do this test. I think the correct location is after the floats.AddScaledTo below.